### DISCUSSION TITLE
Organization folders

### DISCUSSION THREAD

**Discussion by sysadmingithub** [dani-garcia/vaultwarden#4270] [Q&A] [answered]
### Subject of the issue

When creating an organization item, there's the option to select personal folders, which if done, creates a folder inside an organization.

### Deployment environment

* vaultwarden version: 1.30.1

* Install method: Docker image

* Clients used: Web vault, Browser extension

### Steps to reproduce

Have an organization where you're a User, with access to a collection.
Create a folder in your personal vault.
When you create a new item in the organization vault, you have the option to choose a folder from your personal vault.
Select the folder and after the item is created, a folder is also created in the organization.
The folder in the organization is only visible to the person who created it.

### Troubleshooting data

Before creating organization item using personal folder

![image](https://github.com/dani-garcia/vaultwarden/assets/58550827/496099eb-3dd5-42c8-940a-0f63afe79241)

Creating an organization item, using the personal vault folder named "Teste"

![image](https://github.com/dani-garcia/vaultwarden/assets/58550827/6a9a3140-a851-48c1-8c22-ef1c46014cc1)

After creating the organization item, the folder is shown in the organization, with the item.

![image](https://github.com/dani-garcia/vaultwarden/assets/58550827/98204dde-74f8-464f-94d5-13a2e2312bef)

Empty "Teste" folder in personal vault.

![image](https://github.com/dani-garcia/vaultwarden/assets/58550827/aabc5ef3-32d5-4e5b-9376-6eef21c8ab1a)

**Comment by stefan0xC** [comment]
That is [intended behavior](https://bitwarden.com/help/folders/).

> **Folders organize your individual vault** (but can include [shared items](https://bitwarden.com/help/sharing/)) and are unique to you, where collections are shared between members of organizations.

**Reply by stefan0xC** [reply]
I think the confusion arises because the personal vault will show all (even empty) folders, while the organization will only show non-empty folders. That is a quirk of the web-vault. Not sure if fixing this is on the [Bitwarden roadmap](https://community.bitwarden.com/t/bitwarden-roadmap/12865) (maybe if they switch from folders to vault item labels).

**Reply by sysadmingithub** [reply]
The problem here is that you can't create folders in an organization, but when creating an item for your organization you can select a personal folder to store it. That then creates a folder in the organization.

This shouldn't be the case, as folders are for your personal vault ("Folders organize your individual vault") and collections are for an organization vault.

About the empty folders, I created a new folder called "Not empty" in the personal vault, inserted an item in it, switched to the organization vault and it wasn't shown.

New folder in personal vault with item in it.
![image](https://github.com/dani-garcia/vaultwarden/assets/58550827/4d7d3280-841f-4db7-b502-ad6f36b10644)

Not showing in organization vault.
![image](https://github.com/dani-garcia/vaultwarden/assets/58550827/4c13fb8c-a52e-47cb-9034-807eb5911070)

**Reply by BlackDex** [reply, marked as answer]
As @stefan0xC already explained, this is how it works.

All folders created will always be shown when selecting `All Vaults` or `My Vault`.
Only non-empty folders will be shown when you select an `Organization`.

This is done so that you can see all folders you created, empty or not.
Also, folders are never created for an organization, they are always linked to the user.
Users are allowed to _place_ an organization item into a folder, which basically only is a list of cipher uuid's linked to a folder uuid owned by the user uuid.

So, this is exactly how it should work.
And, as @stefan0xC also already mentioned, they seem to be planning for something that is called labels or tags or something. Which would mean you can add multiple labels/tags to a cipher (for you personally, and probably also within a org which then are shared i think). That is kinda the same as folders, but instead of only allowing one tag/label (aka folder) you are probably allowed to add multiple.

**Reply by BlackDex** [reply]
Also, this is not something we can fix on the server side.
We just send a list of which cipher uuid's are linked to which folder uuid's. The clients stitches them together.

**Comment by dkCorpo** [comment]
I just stumbled upon this while managing our vaultwarden instance.
I had use an account with user privileges who's part of only one organization and has the rights to manage only one collection.

User:
created a testfolder and moved one credential from the organization's collection to that folder. it didn't show up in the organization's collection anymore for this user.

Admin:
Still saw the moved item inside the organization's collection without the folder. No folder visible.

This is kinda confusing.
Is there any way to disable folders entirely? We already enabled single organization ownership for organizations only.

**Reply by stefan0xC** [reply]
> User:
> created a testfolder and moved one credential from the organization's collection to that folder. it didn't show up in the organization's collection anymore for this user.

I can't replicate that. The organization item that is saved in a folder stays accessible via the collection:

https://github.com/user-attachments/assets/10c23fb7-391a-48ba-ba57-3fd8176d73f7

> Admin:
> Still saw the moved item inside the organization's collection without the folder. No folder visible.

Folders are not shared, they are [a way to organize your own vault (including shared items)](https://bitwarden.com/help/folders/) so this is expected.

>  Is there any way to disable folders entirely? We already enabled single organization ownership for organizations only.

No, I don't think so.
